### PR TITLE
Keep keys in binary encoding

### PR DESCRIPTION
**Pull request by aravindet**
Currently, the Graffy "plumbing" representation keeps keys (the key and end properties) as strings. String $keys are directly copied over to the key property, while other data types are encoded in a binary array in an order-preserving way, then base64 encoded. To distinguish between encoded and non-encoded keys, encoded keys are also prefixed with a null character.

This scheme has the following issues:
- Encoded keys are unnecessarily long, and undergo extra processing
- There is the complexity of dealing with two "kinds" of keys throughout the code

This PR encodes all $key values uniformly into Uin8Arrays and keeps them in that format in all Graffy core functions, only doing the Base64 encoding for sending over the network. (It is proposed to drop this and switch to sending binary, in a separate PR.)

### CODE REVIEW

File: src/client/httpClient.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ class AggregateQuery {
       }
       return;
     }
-    const data = await response.json();
+    const data = unpack(JSON.parse(await response.text()));
```
Comment (Ashniu123) [inline src/client/httpClient.js]: Shouldn't there be a `reader.reject` in case the `parse`-ing fails?

File: src/common/coding/base64.js [hunk 2]
```diff
@@ -27,7 +28,7 @@ export function encode(u8Arr) {
     str += gstr;
```
Comment (Ashniu123) [inline src/common/coding/base64.js:28]: What does this `encode` do?

File: src/common/coding/args.js [hunk 3]
```diff
@@ -38,14 +41,15 @@ export function splitArgs(arg) {
...
+  if (!isPlainObject(arg)) return { key: encodeValue(arg) };
 
   const [page, filter] = splitArgs(arg);
   errIf('page_and_filter', page && filter, arg);
 
-  if (!page) return { key: maybeEncode(filter || {}) };
+  if (!page) return { key: encodeValue(filter || {}) };
 
   const { $cursor, ...range } = page;
+  // @ts-ignore
```
Comment (Ashniu123) [inline src/common/coding/args.js:52]: random ts-ignore in js project?

File: src/common/coding/args.js [hunk 4]
```diff
@@ -74,32 +78,35 @@ export function encode(arg) {
 }
 
 export function decode(node) {
-  if (typeof node === 'string') return node;
+  if (typeof node === 'string') throw Error('why?');
```
Comment (Ashniu123) [inline src/common/coding/args.js]: Prolly try for a better error message?

File: src/common/coding/base64.js (hunk 2 above)
Comment (aravindet) [reply src/common/coding/base64.js:28]: Converts a Uint8Array into a string using Base64 algorithm, but with my custom alphabet. The custom alphabet has exactly the same letters as the well-known "URL-safe" variant of Base64, but with the characters ordered by ascii value.

- Usual b64url alphabet: `ABC..Zabc...z012...9-_`
- This alphabet: `-012...9ABC...Z_abc...z`

File: src/common/coding/args.js (hunk 3 above)
Comment (aravindet) [reply src/common/coding/args.js:52]: I use Typescript's checkJS option to lint this project. Type definitions are really bad though, and could use some help!

File: src/common/coding/args.js (hunk 4 above)
Comment (aravindet) [reply src/common/coding/args.js]: :rofl: Good catch. This was debugging code I forgot to remove.

**Pull request merged**